src/acquire.c

```diff
@@ -490,6 +507,7 @@ acquire_start(struct AcquireRuntime* self_)
 
         CHECK(video_sink_start(&video->sink, &self->device_manager) ==
               Device_Ok);
+        CHECK(reserve_image_shape(video));
```

Comment:
What do you think about doing this in configure() instead?

I usually use configure() for things like allocation/resource-acquisition so that start() is pretty quick. Also, that way you can rely on the config settings rather than querying the camera for the image shape.

Reply:
It originally was in configure(), but the storage device isn't created until `video_sink_start()`. At that point all you have is the storage properties, which tbh is probably why I chose to put the image shape in the storage properties in the first place. I didn't remember this until my downstream tests started failing. If we're reserving resources around the image shape in the storage device, this is pretty much the only place we can do it. I would consider it "startup behavior" anyway.

Anyway, I'm not happy with the way it looks in the code / the precedent it sets, though. An alternative is to change `video_*_start` to take the entire video_s pointer and move `reserve_image_shape` into `video_sink_start`.